File: TMessagesProj/src/main/java/ua/gomin/messenger/preferences/GominPreferencesNavigator.java
package ua.gomin.messenger.preferences;

import org.telegram.ui.ActionBar.BaseFragment;

/**
 * Навігація між екранами налаштувань Гоміна.
 */
public class GominPreferencesNavigator {

    public static final GominPreferencesNavigator INSTANCE = new GominPreferencesNavigator();

    public void createGominSettings(BaseFragment fragment) {
        fragment.presentFragment(new GominSettingsEntry());
    }

    public void createAbout(BaseFragment fragment) {
        // TODO: AboutPreferencesEntry
    }

    public void createGemini(BaseFragment fragment) {
        // TODO: GeminiPreferencesEntry
    }

    public void createDebug(BaseFragment fragment) {
        // TODO: DebugPreferencesEntry
    }
}

File: TMessagesProj/src/main/java/ua/gomin/messenger/preferences/GominPreferencesNavigator.kt
package ua.gomin.messenger.preferences

import org.telegram.ui.ActionBar.BaseFragment

/**
 * Навігація між екранами налаштувань Гоміна.
 */
object GominPreferencesNavigator {

    fun createGominSettings(fragment: BaseFragment) {
        fragment.presentFragment(GominSettingsEntry())
    }

    fun createAbout(fragment: BaseFragment) {
        // TODO: AboutPreferencesEntry
    }

    fun createGemini(fragment: BaseFragment) {
        // TODO: GeminiPreferencesEntry
    }

    fun createDebug(fragment: BaseFragment) {
        // TODO: DebugPreferencesEntry
    }
}
